src/scripts/components/VideoOverlay.js

```diff
@@ -2,57 +2,148 @@
 import styled from 'styled-components'
-import VideoRecord from 'records/VideoRecords'
 
-import Wrapper from './foundations/Wrapper'
+import VideoRecord from 'records/VideoRecords'
 
 type Props = {
   video: ?VideoRecord,
   match: {
     params: {
       id: String
     }
+  },
+  isEmbed?: boolean,
+  onClick: (e: Object) => void
+}
+
+type State = {
+  openPopover: ?string,
+  buttons: {
+    profile: ?Class<React.Component<any>>
   }
-  // video:{
-  //   title: String
-  // }
 }
 
+const overlayPadding: string = '48px'
+
 const Overlay = styled.div`
   width: 100%;
-  top: 0;
-  left: 0;
+  height: 100%;
+  flex: 0 0 100%;
+  display: flex;
+  flex-direction: column;
   color: white;
   background: rgba(0, 0, 0, 0.8);
+  padding: ${overlayPadding};
+  box-sizing: border-box;
+`
+
+const TopBar = styled.div`
+  display: flex;
+  flex-direction: row;
+`
+
+const Title = styled.div`
+  font-size: 24px;
+  flex: 1 0 50%;
+`
+
+const ButtonGroup = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex: 1 0 0;
+  justify-content: flex-end;
+`
+
+const PopoverWrapper = styled.div`
   position: absolute;
-  z-index: 10;
+  top: ${overlayPadding};
+  right: ${overlayPadding};
+  width: 200px;
+  height: 100px;
+  display: ${props => (props.open ? 'block' : 'none')};
 `
 
-class VideoOverlay extends Component<Props, void> {
-  render () {
-    const videoId = this.props.match.params.id
-    const video = this.props.video
-    if (video === undefined) {
-      return (
-        <Wrapper>
-          <Overlay id="video-overlay">
-            <h2>Video Overlay</h2>
-            <h5>This videow (with id {videoId}) could not be found!</h5>
-          </Overlay>
-        </Wrapper>
-      )
-    } else {
-      return (
-        <Wrapper>
-          <Overlay id="video-overlay">
-            <h2>Video Overlay</h2>
-            <h5>The video id is: {videoId}</h5>
-            Title: {(video && video.title) || 'This video has no (known) title'}
-          </Overlay>
-        </Wrapper>
+class VideoOverlay extends Component<Props, State> {
+  onProfileButtonClick: (e: Object) => void
+  popoverWrapperRefCallback: (ref: HTMLElement) => void
+  popoverWrapperRef: ?HTMLElement
+
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {
+      openPopover: null,
+      buttons: {
+        profile: null
+      }
+    }
+
+    this.loadEmbedPlugins()
+  }
+
+  loadEmbedPlugins () {
+    const { isEmbed } = this.props
+
+    if (isEmbed) {
+      import(/* webpackChunkName: ProfileButton */ 'components/widgets/PlayerPlugins/ProfileButton').then(
+        ProfileButtonModule => {
+          const ProfileButton: Class<
+            React.Component<any>
+          > = ((ProfileButtonModule.default: any): Class<React.Component<any>>)
+          this.setState(prevState => ({
+            buttons: {
+              ...prevState.buttons,
+              profile: ProfileButton
+            }
+          }))
+        }
```

Comment:
Above is probably the most interesting part of this PR. Basically, this instructs webpack to split `ProfileButton.js` into another module and only load it dynamically when we go down this code path (i.e. when we are viewing the player via the `/embed` route. This prevents us from loading these "plugins" unless they are going to be used.